arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -538,17 +593,39 @@ public inline fun <Error, A, B> Raise<Error>.mapOrAccumulate(
 public inline fun <Error, A, B> Raise<NonEmptyList<Error>>.mapOrAccumulate(
   iterable: Iterable<A>,
   @BuilderInference transform: RaiseAccumulate<Error>.(A) -> B
-): List<B> {
-  val error = mutableListOf<Error>()
-  val results = ArrayList<B>(iterable.collectionSizeOrDefault(10))
-  for (item in iterable) {
-    fold<NonEmptyList<Error>, B, Unit>(
-      { transform(RaiseAccumulate(this), item) },
-      { errors -> error.addAll(errors) },
-      { results.add(it) }
-    )
-  }
-  return error.toNonEmptyListOrNull()?.let { raise(it) } ?: results
+): List<B> = buildList(iterable.collectionSizeOrDefault(10)) {
+  forEachAccumulating(iterable) { add(transform(it)) }
+}
+
+/**
+ * Transform every element of [sequence] using the given [transform], or accumulate all the occurred errors using [combine].
+ *
+ * See the Arrow docs for more information over
+ * [error accumulation](https://arrow-kt.io/learn/typed-errors/working-with-typed-errors/#accumulating-errors)
+ * and how to use it in [validation](https://arrow-kt.io/learn/typed-errors/validation/).
+ */
+@RaiseDSL
+public inline fun <Error, A, B> Raise<Error>.mapOrAccumulate(
+  sequence: Sequence<A>,
+  combine: (Error, Error) -> Error,
+  @BuilderInference transform: RaiseAccumulate<Error>.(A) -> B
+): List<B> = buildList {
+  forEachAccumulating(sequence, combine) { add(transform(it)) }
+}
```

Comment:
Note that this is not what one would expect from `Sequence.map`, it should rather return another `Sequence` (but it should be as easy as doing `buildSequence` instead)

Reply:
There's sadly no way to do `mapOrAccumulate` while returning a `Sequence` because to accumulate the errors we must do it eagerly. There's already precedent for this in the codebase with `Sequence.mapOrAccumulate` returning `Either<Error, List<A>>`. I think returning `List` emphasizes that this is a terminal operation. Crazy idea: `RaisingSequence<Error, A>` whose operations need a `Raise<Error>`.

Reply:
You are completely correct, I stand corrected